grafana/src/main/java/org/apache/iotdb/web/grafana/dao/impl/BasicDaoImpl.java

```diff
@@ -104,9 +103,10 @@ public Object doInConnection(Connection connection) throws SQLException {
   public List<TimeValues> querySeries(String s, Pair<ZonedDateTime, ZonedDateTime> timeRange) {
     Long from = zonedCovertToLong(timeRange.left);
     Long to = zonedCovertToLong(timeRange.right);
-    String sql = "SELECT " + s.substring(s.lastIndexOf('.') + 1) + " FROM root."
-        + s.substring(0, s.lastIndexOf('.')) + " WHERE time > " + from * TIMESTAMP_RADIX
-        + " and time < " + to * TIMESTAMP_RADIX;
+    // How many rows will the result have?
```

Comment:
currently, we can not get how many rows before we get the whole ResultSet.
===

 I have an idea, we may implement it by using the summaries (count of each page/chunk) info in IoTDB. 

@qiaojialin , how do you think?

Reply:
do you want to get the precise result or a fuzy one?